cmd/exec.go

```diff
@@ -0,0 +1,196 @@
+	if err != nil {
+		return errors.Wrap(err, "config error")
+	}
+
+	ctx := context.Background()
+	ready := make(chan struct{})
+
+	opa, err := sdk.New(ctx, sdk.Options{
+		Config:        bytes.NewReader(config),
+		Logger:        stdLogger,
+		ConsoleLogger: consoleLogger,
+		Ready:         ready,
+	})
+	if err != nil {
+		return errors.Wrap(err, "runtime error")
+	}
+
+	if err := triggerPlugins(ctx, opa, []string{discovery.Name, bundle.Name, status.Name}); err != nil {
+		return errors.Wrap(err, "runtime error")
+	}
+
+	<-ready
+
+	if err := exec.Exec(ctx, opa, params); err != nil {
+		return err
+	}
+
+	if err := triggerPlugins(ctx, opa, []string{logs.Name}); err != nil {
+		return errors.Wrap(err, "runtime error")
+	}
+
+	return nil
+}
+
+func triggerPlugins(ctx context.Context, opa *sdk.OPA, names []string) error {
+	for _, name := range names {
+		if p, ok := opa.Plugin(name).(plugins.Triggerable); ok {
+			if err := p.Trigger(ctx); err != nil {
+				return errors.Wrap(err, "runtime error")
+			}
+		}
+	}
+	return nil
+}
+
+func setupLogging(level, format string) (logging.Logger, logging.Logger, error) {
+
+	lvl, err := internal_logging.GetLevel(level)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	logging.Get().SetFormatter(internal_logging.GetFormatter(format))
+	logging.Get().SetLevel(lvl)
+
+	stdLogger := logging.New()
+	stdLogger.SetLevel(lvl)
+	stdLogger.SetFormatter(internal_logging.GetFormatter(format))
+
+	consoleLogger := logging.New()
+	consoleLogger.SetFormatter(internal_logging.GetFormatter(format))
+
+	return stdLogger, consoleLogger, nil
+}
+
+func setupConfig(file string, overrides []string, overrideFiles []string) ([]byte, error) {
+
+	bs, err := config.Load(file, overrides, overrideFiles)
+	if err != nil {
+		return nil, errors.Wrap(err, "config error")
+	}
+
+	var root map[string]interface{}
+
+	if err := util.Unmarshal(bs, &root); err != nil {
+		return nil, err
+	}
+
+	// NOTE(tsandall): This could be generalized in the future if we need to
+	// deal with arbitrary plugins.
+
+	// NOTE(tsandall): Overriding the discovery trigger mode to manual means
+	// that all plugins will inherit the trigger mode by default. If the plugin
+	// trigger mode is explicitly set to something other than 'manual' this will
+	// result in a configuration error.
+	if cfg, ok := root["discovery"].(map[string]interface{}); ok {
+		cfg["trigger"] = "manual"
+	}
+
+	if cfg, ok := root["bundles"].(map[string]interface{}); ok {
+		for _, x := range cfg {
+			if bcfg, ok := x.(map[string]interface{}); ok {
+				bcfg["trigger"] = "manual"
+			}
+		}
+	}
+
+	if cfg, ok := root["decision_logs"].(map[string]interface{}); ok {
+		if rcfg, ok := cfg["reporting"].(map[string]interface{}); ok {
+			rcfg["trigger"] = "manual"
+		}
+	}
+
+	if cfg, ok := root["status"].(map[string]interface{}); ok {
+		cfg["trigger"] = "manual"
+	}
+
+	return util.MustMarshalJSON(root), nil
```

Comment:
```suggestion
	return json.Marshal(root)
```

Let's avoid the panic. If we'd rather have a util method, let's add `util.MarshalJSON(interface{}) ([]byte, error)`.